x-pack/plugin/deprecation/src/main/java/org/elasticsearch/xpack/deprecation/ClusterDeprecationChecks.java

```diff
@@ -99,7 +101,57 @@ static DeprecationIssue checkUserAgentPipelines(ClusterState state) {
                 "Ingest pipelines " + pipelinesWithDeprecatedEcsConfig + " will change to using ECS output format by default in 7.0");
         }
         return null;
+    }
+
+    @SuppressWarnings("unchecked")
+    static DeprecationIssue checkFormatOnPipeline(ClusterState state) {
+        List<PipelineConfiguration> pipelines = IngestService.getPipelines(state);
+
+        StringJoiner pipelinesMessages = new StringJoiner("\n");
+        for (PipelineConfiguration pipeline : pipelines) {
+            Map<String, Object> pipelineConfig = pipeline.getConfigAsMap();
+
+            List<Map<String, Map<String, Object>>> processors =
+                (List<Map<String, Map<String, Object>>>) pipelineConfig.get("processors");
+
+            List<String> fields = new ArrayList<>();
+            for (Map<String, Map<String, Object>> processor : processors) {
+                if (processor.containsKey("date")) {
+                    Map<String, Object> date = processor.get("date");
+                    List<String> formats = (List<String>) date.get("formats");
+                    for (String format : formats) {
+                        if (JodaDeprecationPatterns.isDeprecatedPattern(format)) {
+                            fields.add(formatPipelineField("date", format));
+                        }
+                    }
+                }
+                if (processor.containsKey("date_index_name")) {
+                    Map<String, Object> date_index_name = processor.get("date_index_name");
+                    String index_name_format = (String) date_index_name.get("index_name_format");
+                    if (JodaDeprecationPatterns.isDeprecatedPattern(index_name_format)) {
+                        fields.add(formatPipelineField("index_name_format", index_name_format));
+                    }
+                }
+
+                if (fields.isEmpty() == false) {
+                    pipelinesMessages.add("pipelineId: " + pipeline.getId() + ", fields: " + fields + ".");
+                }
+            }
+        }
+
+        if (pipelinesMessages.length() != 0) {
+            return new DeprecationIssue(DeprecationIssue.Level.WARNING,
+                "Pipelines contain date fields with deprecated format",
+                "https://www.elastic.co/guide/en/elasticsearch/reference/7.0/breaking-changes-7.0.html#breaking_70_java_time_changes",
+                "Ingest pipelines contain deprecated joda patterns: " + pipelinesMessages);
```

Comment:
I think we should make this similar to the message that's used by the index-level check, like:

```suggestion
                "Ingest pipelines which contain deprecated date formats: " + pipelinesMessages);
```